Terminal.Gui/View/Layout/Dim.cs

```diff
@@ -0,0 +1,766 @@
+                // TODO: AnchorEnd needs work
+                // TODO: If _min > 0 we can SetRelativeLayout for the subviews?
+                subviewsSize = 0;
+
+                if (us.Subviews.Count > 0)
+                {
+                    for (var i = 0; i < us.Subviews.Count; i++)
+                    {
+                        View v = us.Subviews [i];
+                        bool isNotPosAnchorEnd = dimension == Dimension.Width ? v.X is not PosAnchorEnd : v.Y is not PosAnchorEnd;
+
+                        //if (!isNotPosAnchorEnd)
+                        //{
+                        //    v.SetRelativeLayout(dimension == Dimension.Width ? (new Size (autoMin, 0)) : new Size (0, autoMin));
+                        //}
+
+                        if (isNotPosAnchorEnd)
+                        {
+                            int size = dimension == Dimension.Width ? v.Frame.X + v.Frame.Width : v.Frame.Y + v.Frame.Height;
+
+                            if (size > subviewsSize)
+                            {
+                                subviewsSize = size;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // All sizes here are content-relative; ignoring adornments.
+        // We take the larger of text and content.
+        int max = int.Max (textSize, subviewsSize);
+
+        // And, if min: is set, it wins if larger
+        max = int.Max (max, autoMin);
+
+        // Factor in adornments
+        Thickness thickness = us.GetAdornmentsThickness ();
+
+        if (dimension == Dimension.Width)
+        {
+            max += thickness.Horizontal;
+        }
+        else
+        {
+            max += thickness.Vertical;
+        }
+
+        // If max: is set, clamp the return - BUGBUG: Not tested
+        return int.Min (max, MaximumContentDim?.Anchor (superviewContentSize) ?? max);
+    }
+
+    internal override bool ReferencesOtherViews ()
+    {
+        // BUGBUG: This is not correct. _contentSize may be null.
+        return false; //_style.HasFlag (DimAutoStyle.Content);
+    }
+}
+
+/// <summary>
+///     Represents a dimension that is a combination of two other dimensions.
+/// </summary>
+/// <param name="add">
+///     Indicates whether the two dimensions are added or subtracted. If <see langword="true"/>, the dimensions are added,
+///     otherwise they are subtracted.
+/// </param>
+/// <remarks>
+///     This is a low-level API that is typically used internally by the layout system. Use the various static
+///     methods on the <see cref="Dim"/> class to create <see cref="Dim"/> objects instead.
+/// </remarks>
+/// <param name="left">The left dimension.</param>
+/// <param name="right">The right dimension.</param>
+public class DimCombine (bool add, Dim left, Dim right) : Dim
+{
+    /// <summary>
+    ///     Gets whether the two dimensions are added or subtracted.
+    /// </summary>
+    public bool Add { get; } = add;
```

Comment:
Should be an enum (default or byte-backed, but unspecified - thus int - is usually best, performance-wise) with 2 members where the 0 value is ok to be named the desired default behavior.

Requires a whole lot less explanation of the behavior when the code is expressive for no extra cost.

If it feels weird to write a boolean as "IsX" or "DoesX" or "HasX" or whatever, a boolean may not be the best choice, even if the enum is simply 2 values.